spyder/plugins/application/widgets/install.py

```diff
@@ -0,0 +1,252 @@
+
+    # Signals
+    # Signal to get the current status of the update installation
+    # str: Status string
+    sig_installation_status = Signal(str)
+
+    def __init__(self, parent):
+
+        self.cancelled = False
+        self.status = NO_STATUS
+        self.thread_install_update = None
+        super(UpdateInstallerDialog, self).__init__(parent)
+        self.setStyleSheet(
+            f"background-color: {QStylePalette.COLOR_BACKGROUND_2}")
+        self.setWindowFlags(Qt.Dialog | Qt.MSWindowsFixedSizeDialogHint)
+        self._parent = parent
+        self._installation_widget = UpdateInstallation(self)
+
+        # Layout
+        installer_layout = QVBoxLayout()
+        installer_layout.addWidget(self._installation_widget)
+
+        self.setLayout(installer_layout)
+
+        # Signals
+        self.sig_download_progress.connect(
+            self._installation_widget.update_installation_progress)
+        self.sig_installation_status.connect(
+            self._installation_widget.update_installation_status)
+        self.sig_installation_status.connect(
+            self.finished_installation)
+
+        self._installation_widget.ok_button.clicked.connect(
+            self.close_installer)
+        self._installation_widget.cancel_button.clicked.connect(
+            self.cancel_install)
+
+        # Show integration widget
+        self.setup()
+
+    def setup(self, installation=False):
+        """Setup visibility of widgets."""
+        self._installation_widget.setVisible(True)
+        self.adjustSize()
+
+    def cancel_install(self):
+        """Cancel the installation in progress."""
+        reply = QMessageBox.critical(
+            self._parent, 'Spyder',
+            _('Do you really want to cancel installing the Spyder update?'),
+            QMessageBox.Yes, QMessageBox.No)
+        if reply == QMessageBox.Yes:
+            self.cancelled = True
+            self.cancell_thread_install_update()
+            self.setup()
+            self.accept()
+            return True
+        return False
+
+    def continue_install(self):
+        """Continue the installation in progress
+        by downloading the installer."""
+        reply = QMessageBox(icon=QMessageBox.Question,
+                            text=_('Do you want to download and'
+                                   ' install the latest version of'
+                                   ' spyder?<br>'),
+                            parent=self._parent)
+        reply.setWindowTitle("Spyder")
+        reply.setAttribute(Qt.WA_ShowWithoutActivating)
+        reply.setStandardButtons(QMessageBox.Yes | QMessageBox.No)
+        reply.buttonClicked.connect(
+                            lambda button:
+                            self.start_installation_update()
+                            if button.text() in ('&Yes')
+                            else self._change_update_installation_status(
+                                status=PENDING))
+        reply.show()
```

Comment:
A similar approach as the one described at `container.py` could be used here (changing the usage of `show` for `exec_` and `result`)